Bottom: 3566ceef05393923c17ec88a04bf72003f065f3f
Top:    827be148af8fc3440ec93bb5d5ffd990b564ba7e
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-07 01:46:11 +0000

Tidy up range check in misc.c

For consistnecy with other range checks involving max_copper_layer, and to
drive home the fact we are checking for exceeding max_copper_layer + 2.
NB: The +2 relates to the bottom silk and top-silk layers, whose numbers
are used to denote the bottom and top layer groups respectively.

re-write:
  layer > max_copper_layer + 1

to become:
  layer >= max_copper_layer + 2


---

diff --git a/src/misc.c b/src/misc.c
index c3e6af0..040124c 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -2050,7 +2050,7 @@ MoveLayerToGroup (int layer, int group)
 {
   int prev, i, j;
 
-  if (layer < 0 || layer > max_copper_layer + 1)
+  if (layer < 0 || layer >= max_copper_layer + 2)
     return -1;
   prev = GetLayerGroupNumberByNumber (layer);
   if ((layer == bottom_silk_layer
